flash/image/embedding/strategies/default.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Optional
+
+import torch
+
+from flash.core.adapter import Adapter, AdapterTask
+from flash.core.data.io.input import DataKeys
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.url_error import catch_url_error
+from flash.image.embedding.heads import IMAGE_EMBEDDER_HEADS
+from flash.image.embedding.losses import IMAGE_EMBEDDER_LOSS_FUNCTIONS
+
+
+class DefaultAdapter(Adapter):
+    """The ``DefaultAdapter`` is an :class:`~flash.core.adapter.Adapter`."""
+
+    required_extras: str = "image"
+
+    def __init__(self, backbone: torch.nn.Module, head: torch.nn.Module):
+        super().__init__()
+
+        self.backbone = backbone
+        self.head = head
+
+    @classmethod
+    @catch_url_error
+    def from_task(
+        cls,
+        *args,
+        task: AdapterTask,
+        backbone: torch.nn.Module,
+        head: torch.nn.Module,
+        **kwargs,
+    ) -> Adapter:
+        adapter = cls(backbone, head)
+        adapter.__dict__["_task"] = task
+        return adapter
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DataKeys.INPUT], batch[DataKeys.TARGET])
+        return Task.training_step(self._task, batch, batch_idx)
```

Comment:
Let's instead raise an error here as we only support training the embedder if using the VISSL integration. Can do something similar to what we have here: https://github.com/PyTorchLightning/lightning-flash/blob/167f5e73fe28630830e15ed40502979a29d38a68/flash/text/embedding/model.py#L81

Reply:
I fixed it. Please review again!